File: shared/src/commonMain/kotlin/com/divya/data/models/Panchang.kt
package com.divya.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Panchang(
    val date: String,
    val timezone: String,
    val tithi: Tithi,
    val nakshatra: Nakshatra,
    val sunriseUtc: String,
    val sunsetUtc: String,
    val rahuKaalStartUtc: String,
    val rahuKaalEndUtc: String,
    val sunriseLocal: String? = null,
    val sunsetLocal: String? = null,
    val rahuKaalStartLocal: String? = null,
    val rahuKaalEndLocal: String? = null,
    val festivals: List<Festival> = emptyList(),
    val infoTooltip: String? = null,
)

@Serializable
data class Tithi(
    val name: String,
    val paksha: String,
    val number: Int,
)

@Serializable
data class Nakshatra(
    val number: Int,
    val name: String,
    val nameHi: String,
    val deity: String,
)

File: backend/src/models/Panchang.js
import mongoose from "mongoose";

const panchangSchema = new mongoose.Schema({
  date: { type: String, required: true, unique: true },
  timezone: { type: String, default: "Asia/Kolkata" },
  tithi: {
    name: String,
    paksha: String,
    number: Number
  },
  nakshatra: {
    name: String,
    nameHi: String,
    deity: String,
    number: Number
  },
  sunriseUtc: String,
  sunsetUtc: String,
  rahuKaalStartUtc: String,
  rahuKaalEndUtc: String,
  festivals: [{ type: mongoose.Schema.Types.Mixed }],
  dailyGuidance: {
    overall: {
      type: String,
      enum: ["highly_auspicious", "auspicious", "neutral", "inauspicious"],
      default: "neutral"
    },
    goodFor: [{ type: String }],
    avoidFor: [{ type: String }],
    auspiciousWindow: String,
    rahuKaalWarning: String
  },
  referenceLocation: {
    lat: { type: Number, default: 9.0167 },
    lng: { type: Number, default: 76.5333 }
  },
  generatedAt: { type: Date, default: Date.now }
});

export const Panchang = mongoose.model("Panchang", panchangSchema);
